src/index/index.c

```diff
@@ -251,13 +283,21 @@ void Index_Construct
 	if(idx->type == IDX_FULLTEXT) {
 		for(uint i = 0; i < fields_count; i++) {
 			// introduce text field
-			RediSearch_CreateTextField(rsIdx, idx->fields[i]);
+			unsigned options = RSFLDOPT_NONE;
+			if(idx->fields[i].phonetic) {
+				options |= RSFLDOPT_TXTPHONETIC;
+			}
+			if(idx->fields[i].nostem) {
+				options |= RSFLDOPT_TXTNOSTEM;
+			}
+			RSFieldID fieldID = RediSearch_CreateField(rsIdx, idx->fields[i].name, RSFLDTYPE_FULLTEXT, options);
+			RediSearch_TextFieldSetWeight(rsIdx, fieldID, idx->fields[i].weight);
```

Comment:
What's the default `weight` value ? not sure what are the consequences of 0 `weight`

Reply:
the default is 1